### ISSUE TITLE
Eligibility check error on iOS

### ISSUE THREAD

**Issue by hasimyerlikaya** [qonversion/flutter-sdk#371]
Hi,
I'm getting the following error when I call `checkTrialIntroEligibility`
Is this related to the package, or is there any problem with Apple?

`❌ Request failed: https://api.qonversion.io/v3/users/.../offers/40_off/signatures, error: Error Domain=com.qonversion.io.api Code=3 "Failed to verify eligibility: failed to verify receipt: could not connect to app store server: Post "https://sandbox.itunes.apple.com/verifyReceipt": context deadline exceeded" UserInfo={NSLocalizedDescription=Failed to verify eligibility: failed to verify receipt: could not connect to app store server: Post "https://sandbox.itunes.apple.com/verifyReceipt": context deadline exceeded}
`

`flutter: PlatformException(BackendError, Failed to verify eligibility: failed to verify receipt: could not connect to app store server: Post "https://sandbox.itunes.apple.com/verifyReceipt": context deadline exceeded, Qonversion Error Code: BackendError, null)
`

**Comment by gorgeoussamq** [comment]
Hi @hasimyerlikaya can you please reach us via support@qonversion.io so we can check your log files?

**Comment by hasimyerlikaya** [comment]
Hi, I haven't seen this error again. Therefore, I'm closing this issue for now.
If I come across this error again, I will let you know.

**Comment by SpertsyanKM** [comment]
Okay, thank you! Have a good day!

**Comment by hasimyerlikaya** [comment]
Issue: checkTrialIntroEligibility Returns Only nonIntroOrTrialProduct on iOS

Description

I am facing an issue with the checkTrialIntroEligibility function on iOS. No matter how many times I test it, the function always returns the nonIntroOrTrialProduct value for all test accounts and all subscriptions.

This behavior prevents me from determining whether to show an intro price to the user. Previously, when I called this function, I encountered an error (as mentioned below), but now it no longer throws any error and simply returns the same value.

Expected Behavior

When calling checkTrialIntroEligibility with a new sandbox user, it should return eligibility for introductory offers.

Actual Behavior
	•	The function consistently returns nonIntroOrTrialProduct regardless of the account or subscription state.
	•	When attempting to purchase a subscription, the native screen shows the intro price, but the function still returns nonIntroOrTrialProduct.
	•	After successfully purchasing a subscription, calling the function again still returns nonIntroOrTrialProduct.


Am I overlooking something in the implementation or sandbox setup? 
Any guidance would be appreciated.

Thank you!
